### DISCUSSION TITLE
Pyright supports pydantic [VSCode] [pylance]

### DISCUSSION THREAD

**Discussion by koxudaxi** [pydantic/pydantic#2698] [General]
Microsoft has started supporting pydantic in [pyright](https://github.com/microsoft/pyright).
I guess the feature will be used in VS Code via [pylance](https://github.com/microsoft/pylance-release).

https://github.com/microsoft/pyright/blob/master/specs/dataclass_transforms.md

I listened to a [podcast](https://talkpython.fm/episodes/show/313/automate-your-data-exchange-with-pydantic) which @samuelcolvin talked about pydantic.
He said he got an email about pydantic from Microsft `pyright` developer.
And  I found the spec page.

I use PyCharm with [pydantic plugin](https://plugins.jetbrains.com/plugin/12861-pydantic). (Also, I develop the plugin)
But, Some users want to support pydantic in VS Code.
I know @tiangolo is interested in this topic.
https://github.com/microsoft/python-language-server/issues/1898

I hope to share this topic in this cominity.

**Comment by tiangolo** [comment]
This is amazing! Thanks for sharing it @koxudaxi !

And thanks for all your work with the PyCharm plug-in. Having great support for pydantic in both editors would be awesome. 🎉
